src/workerd/server/server.c++

```diff
@@ -4322,6 +4322,9 @@ kj::Promise<bool> Server::test(jsg::V8System& v8System,
       ++failCount;
     }
     KJ_LOG(DBG, kj::str(result ? "[ PASS ] "_kj : "[ FAIL ] "_kj, name));
+
+    // Wait until everything on this thread is wrapped up.
+    co_await kj::yieldUntilWouldSleep();
```

Comment:
Is this ok? I'm not sure if maybe `finishedPendingEventsFulfiller` should actually be fulfilled before waiting for the thread to become idle. Thoughts @kentonv?